src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -303,6 +305,14 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			if (contents) {
 				this._defaultLanguageInfo = contents.metadata && contents.metadata.language_info;
 				this._savedKernelInfo = this.getSavedKernelInfo(contents);
+				if (contents.metadata) {
+					//Telemetry of loading notebook
+					if (contents.metadata.azdata_notebook_guid) {
+						this._adstelemetryService.createActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.TelemetryAction.Open)
+							.withAdditionalProperties({ azdata_notebook_guid: contents.metadata.azdata_notebook_guid })
```

Comment:
As discussed with Wolfgang - please do validation to ensure this is a valid GUID before sending it. We don't want to just send any value in here in case it got modified for some reason

Reply:
Done